lib/src/signed_video_h26x_auth.c

```diff
@@ -883,23 +874,18 @@ prepare_for_validation(signed_video_t *self)
   svrc_t status = SV_UNKNOWN_FAILURE;
   SV_TRY()
     h26x_nalu_list_item_t *sei = h26x_nalu_list_get_next_sei_item(nalu_list);
-    if (sei && !sei->has_been_decoded) {
+    if (sei) {
       // Decode the SEI and set signature->hash
-      self->latest_validation->public_key_has_changed = false;
-      const uint8_t *tlv_data = sei->nalu->tlv_data;
-      size_t tlv_size = sei->nalu->tlv_size;
-
-      SV_THROW(decode_sei_data(self, tlv_data, tlv_size));
-      sei->has_been_decoded = true;
-      if (self->gop_info->signature_hash_type == DOCUMENT_HASH) {
+      if (!sei->has_been_decoded) {
+        self->latest_validation->public_key_has_changed = false;
+        const uint8_t *tlv_data = sei->nalu->tlv_data;
+        size_t tlv_size = sei->nalu->tlv_size;
+        SV_THROW(decode_sei_data(self, tlv_data, tlv_size));
+        sei->has_been_decoded = true;
         memcpy(verify_data->hash, sei->hash, hash_size);
       }
-    } else if (sei && sei->has_been_decoded) {
-      detect_sei_loss_and_sync_gop(self);
-    }
-
-    if (sei) {
       SV_THROW(prepare_for_link_and_gop_hash_verification(self, sei));
+      detect_lost_sei_and_update_gop_state(self);
```

Comment:
You have changed the order of these two operations. Is it more correct to prepare for link and gop hash verification before we know if we detect a lost sei?